### ISSUE TITLE
Problems programing reader

### ISSUE THREAD

**Issue by rusteff** [maddox/magic-cards#26]
When i run the software and try to program the reader i get this output.
Seams like the reader is not recognized as a programmable USB device on my Windows 10 computer.
I got the 125KHZ RFID EM4100 ID Card Reader.

ÎÞÓÐÐ§Êý¾Ý£¬Çë¼ì²éÉè±¸ÊÇ·ñÁª½Ó....
Data error£¬no data recevice..   06:31:46 PM

**Comment by Velly56** [comment]
+1 Same problem on Win7, winXP etc.  Seems that our reader is not compatible with software..

**Comment by dnguyen800** [comment]
I get the same issue on Windows 10, though if you open Notepad and scan an ID card, the reader does output some numbers. So the card reader works, but not with the software mentioned in the tutorial.

**Comment by rusteff** [comment]
Anyone have a link to a working reader?

**Comment by dnguyen800** [comment]
Yes, I do, but it is the same USB reader I initially had trouble with because the reader did not work with the RFID Reader software mentioned in the Magic-Cards instructions. I learned the magic-cards software works as long as the reader can output some numbers, so I skipped the step 'Programming the Reader' and finished the project. I think the reader was already configured to output in the 10 no. in Hex.

My recommendation is you should test your USB card reader again. Connect USB reader to your computer, open up a text editor, scan an RFID card and see if a 10-digit number appears in the text editor. If it does, then your reader should work.

Here is the eBay item I bought, though the seller I purchased from is no longer selling (his feedback profile looks suspiciously positive)
https://www.ebay.com/p/125khz-USB-RFID-Contactless-Proximity-Sensor-Smart-ID-Card-Reader-Em4100/501755355?iid=253905865944

I think you will just have to try your luck at finding the right one, or try testing the one you already have and see if it outputs a unique number. You may not need to program the USB reader.

**Comment by Velly56** [comment]
dnguyen800, sincere thanks! It is exactly like you said! Chapeaux!

**Comment by DavidDeSloovere** [comment]
Also getting **Data error£¬no data recevice.**
My codes look more like this:

```
àà&(çèè"(é
àà&(çàé'!§
àà&(ç&'à'§
àà&(!èé!ç(
àà&(çèè"(é
```

Should they be actual digits in notepad?

**Comment by dnguyen800** [comment]
> Also getting **Data error£¬no data recevice.**
> My codes look more like this:
> 
> ```
> àà&(çèè"(é
> àà&(çàé'!§
> àà&(ç&'à'§
> àà&(!èé!ç(
> àà&(çèè"(é
> ```
> 
> Should they be actual digits in notepad?

Yes, it should be digits in Notepad. You can use the RFID reader software listed in the [documentation](https://github.com/maddox/magic-cards/blob/master/docs/hardware.md) and hope that the reader is reconfigured to  '10 no. in Hex'. Even though I received an error when using the software, my RFID reader was able to print a 10 digit number in Notepad. I'm not sure if it was able to do it before using the software, but it works now.

**Comment by DavidDeSloovere** [comment]
I can not get it to work on Windows. Can't reconfigure the way the reader reads the ID.
But... it works with the magic cards docker image though. I can see the digits in the logs of the image.
So just an inconvenience really.

**Comment by micolombier** [comment]
@dnguyen800 thanks for your feedback. I am having the same issue. So I will just go on with the project and finger crossed everything will work.

@DavidDeSloovere The codes you had were basically because your caps lock was not enabled. Readers are made for qwerty keyboards. On these keyboards, when you hit the keys above the letters, it types the numbers first and not the special character.
So you just have to enable caps lock when you test and you will see the numbers appearing :)

**Comment by DavidDeSloovere** [comment]
@micolombier Thanks for your insight on the keyboard layout. Makes total sense. 👍
I can't believe I didn't figure this out myself. On the other hand, the reader software could just alter the input.

**Comment by Netizendev** [comment]
ÎÞÓÐÐ§Êý¾Ý£¬Çë¼ì²éÉè±¸ÊÇ·ñÁª½Ó....
Data error£¬no data recevice..

The program is called demo. Is there a full verison out there? dl links?
